gitcredentialhelper/gitcredentialhelper.go

```diff
@@ -0,0 +1,168 @@
+package gitcredentialhelper
```

Comment:
Can we please avoid adding one more top level package. Instead I would move this package to `cmd/argocd-util/commands/repocreds`

Reply:
Thanks for this suggestion.  I agree: if this isn't to be a top-level command then it doesn't warrant a top-level package.

However, there are no other code directories in `cmd/argocd-util/commands`, so adding one for this feels weird...

I have instead moved it into the `util/git` package.  I'm not 100% that it belongs here, though, or whether it should be in its own `util/` package.  @alexmt, what do you think?